src/plugins/intel_gna/src/transformations/gather_sinking_utils.cpp

```diff
@@ -0,0 +1,285 @@
+
+using namespace ov;
+using namespace ov::opset9;
+
+using NodePtr = std::shared_ptr<Node>;
+
+GatherInputsInfo GetFirstGatherInput(NodePtr node) {
+    for (size_t input_idx = 0; input_idx < node->get_input_size(); ++input_idx) {
+        NodePtr input_node = node->get_input_node_shared_ptr(input_idx);
+        auto gather_node = as_type_ptr<Gather>(input_node);
+        if (!gather_node)
+            continue;
+        auto indices_const_node = as_type_ptr<Constant>(gather_node->input_value(1).get_node_shared_ptr());
+        if (!indices_const_node)
+            continue;
+        auto axes_const_node = as_type_ptr<Constant>(gather_node->input_value(2).get_node_shared_ptr());
+        if (!axes_const_node)
+            continue;
+        {
+            GatherInputsInfo input_info;
+            input_info.gather = gather_node;
+            input_info.indices_const = indices_const_node;
+            input_info.axes_const = axes_const_node;
+            input_info.input_idx = input_idx;
+            return input_info;
+        }
+    }
+
+    return GatherInputsInfo();
+}
+
+bool IfNodeHasGatherInputs(const Output<Node>& output) {
+    GatherInputsInfo inputs_info = GetFirstGatherInput(output.get_node_shared_ptr());
+    return !inputs_info.isEmpty();
+}
+
+namespace {
+
+bool HasDynamicRankInput(NodePtr node) {
+    for (auto& input_node : node->input_values()) {
+        const ov::Rank output_rank = input_node.get_partial_shape().rank();
+        if (output_rank.is_dynamic())
+            return true;
+    }
+    return false;
+}
+
+ov::Rank::value_type GetMaxInputRank(const NodePtr& node) {
+    ov::Rank::value_type max_input_rank = 0;
+    for (auto& input_node : node->input_values()) {
+        const ov::Rank output_rank = input_node.get_partial_shape().rank();
+        if (output_rank.is_dynamic())
+            return -1;
+        const ov::Rank::value_type output_rank_len = output_rank.get_length();
+        if (output_rank_len > max_input_rank)
+            max_input_rank = output_rank_len;
+    }
+    return max_input_rank;
+}
+
+NodePtr InsertUnsqueeze(Output<Node> node, size_t n_dims) {
+    std::vector<size_t> dims(n_dims);
+    std::iota(dims.begin(), dims.end(), 0);
+    auto unsqueeze_const = std::make_shared<Constant>(ov::element::i64, Shape{dims.size()}, dims);
+    auto unsqueeze = std::make_shared<Unsqueeze>(node, unsqueeze_const);
+    copy_runtime_info(node.get_node_shared_ptr(), {unsqueeze, unsqueeze_const});
+    return unsqueeze;
+}
+
+ov::Output<ov::Node> FixInputNodeRank(ov::Output<ov::Node> input_node, ov::Rank::value_type required_rank) {
+    const ov::Rank::value_type output_rank = input_node.get_partial_shape().rank().get_length();
+    if (output_rank >= required_rank)
+        return input_node;
+    return InsertUnsqueeze(input_node, required_rank - output_rank)->output(0);
+}
+
+ov::AxisVector ReverseGatherIndexes(const ov::AxisVector & indexes) {
+    ov::AxisVector positions(indexes.size());
```

Comment:
Why it cannot be copied and reversed:
 ov::AxisVector inverted(indexes.rbegin(), indexes.rend());

Reply:
used your way